File: book/petrol-pump/java/chapter4/section9/KeypadPump.java
package chapter4.section9;

import pump.*;
import nz.sodium.*;
import java.util.Optional;

public class KeypadPump implements Pump
{
    public Outputs create(Inputs inputs) {
        Keypad ke = new Keypad(inputs.sKeypad, new Stream<Unit>());
        return new Outputs()
            .setPresetLCD(ke.value.map(v ->
                Formatters.formatSaleCost((double)v)))
            .setBeep(ke.sBeep);
    }
}

File: book/petrol-pump/c#/PetrolPump/Chapter4/Section9/KeypadPump.cs
using Sodium.Frp;
using Sodium.Functional;

namespace PetrolPump.Chapter4.Section9
{
    public class KeypadPump : IPump
    {
        public Outputs Create(Inputs inputs)
        {
            Keypad ke = new Keypad(inputs.SKeypad, Stream.Never<Unit>());
            return new Outputs()
                .SetPresetLcd(ke.Value.Map(Formatters.FormatPresetAmount))
                .SetBeep(ke.SBeep);
        }
    }
}